crates/eth-rpc/src/rpc.rs

```diff
@@ -0,0 +1,42 @@
+use jsonrpsee::RpcModule;
+use kakarot_rpc_core::client::config::{JsonRpcClientBuilder, StarknetConfig};
+use kakarot_rpc_core::client::KakarotClient;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+
+use crate::api::eth_api::EthApiServer;
+use crate::api::kakarot_custom_api::KakarotCustomApiServer;
+use crate::servers::eth_rpc;
+use crate::servers::eth_rpc::KakarotEthRpc;
+use crate::servers::kakarot_custom_rpc::{self, KakarotCustomRpc};
+
+pub struct KakarotRpc {
+    eth_rpc_module: RpcModule<KakarotEthRpc<JsonRpcClient<HttpTransport>>>,
+    kakarot_custom_rpc_module: RpcModule<KakarotCustomRpc<JsonRpcClient<HttpTransport>>>,
```

Comment:
Same here, especially if they use the same type (`JsonRpcClient<HttpTransport>`), it could be passed as a generic to `KakarotRpc`

Reply:
This is also now generic over `Provider`.

<img width="788" alt="image" src="https://github.com/kkrt-labs/kakarot-rpc/assets/41180869/7796e2f1-c56d-4bd7-8817-ace6d78d74dd">